db/migrate/20240118102648_add_position_to_exercise_approaches.rb

```diff
@@ -0,0 +1,13 @@
+class AddPositionToExerciseApproaches < ActiveRecord::Migration[7.0]
+  def change
+    return if Rails.env.production?
+
+    add_column :exercise_approaches, :position, :integer, null: true, limit: 1
+    add_index :exercise_approaches, [:exercise_id, :position]
+
+    # The positions will be fixed manually by running Git::SyncTrack.()
+    Exercise::Approach.update_all(position: 0)
+
+    change_column_null :exercise_approaches, :position, false
```

Comment:
You could also do:

```suggestion
    add_column :exercise_approaches, :position, :integer, null: false, limit: 2, default: 0
    add_index :exercise_approaches, [:exercise_id, :position]
```

Reply:
Yes, but then it would silently fail if our code ever forgot to pass in the position

Reply:
I mean, I don't massively care, but 🤷